---
string actionStatement
string[] actionPredicates
string[] actionFunctions
string stateStatement
string[] statePredicates
string[] stateFunctions
string goalStatement